samples/pubsub.py

```diff
@@ -109,17 +110,31 @@ def on_message_received(topic, payload, dup, qos, retain, **kwargs):
             keep_alive_secs=6)
 
     else:
-        mqtt_connection = mqtt_connection_builder.mtls_from_path(
-            endpoint=args.endpoint,
-            cert_filepath=args.cert,
-            pri_key_filepath=args.key,
-            client_bootstrap=client_bootstrap,
-            ca_filepath=args.root_ca,
-            on_connection_interrupted=on_connection_interrupted,
-            on_connection_resumed=on_connection_resumed,
-            client_id=args.client_id,
-            clean_session=False,
-            keep_alive_secs=6)
+        if args.port:
+            mqtt_connection = mqtt_connection_builder.mtls_from_path(
+                endpoint=args.endpoint,
+                port=args.port,
+                cert_filepath=args.cert,
+                pri_key_filepath=args.key,
+                client_bootstrap=client_bootstrap,
+                ca_filepath=args.root_ca,
+                on_connection_interrupted=on_connection_interrupted,
+                on_connection_resumed=on_connection_resumed,
+                client_id=args.client_id,
+                clean_session=False,
+                keep_alive_secs=6)
+        else:
+            mqtt_connection = mqtt_connection_builder.mtls_from_path(
+                endpoint=args.endpoint,
+                cert_filepath=args.cert,
+                pri_key_filepath=args.key,
+                client_bootstrap=client_bootstrap,
+                ca_filepath=args.root_ca,
+                on_connection_interrupted=on_connection_interrupted,
+                on_connection_resumed=on_connection_resumed,
+                client_id=args.client_id,
+                clean_session=False,
+                keep_alive_secs=6)
```

Comment:
Let's try to keep a single call to `mqtt_connection_builder.mtls_from_path()`:
```
# use port from args, or pass None to get default
port = args.port if args.port else None

mqtt_connection = mqtt_connection_builder.mtls_from_path(
    ...
    port=port
    ...)

```

Reply:
This requires also changes to the this https://github.com/aws/aws-iot-device-sdk-python-v2/blob/7bd47b56226352ce815d82c8f8f0fd3ab8006fae/awsiot/mqtt_connection_builder.py#L153

With the current solution, port is not defined, and `get('port', port)` results in `port`. If I set `port` to None, the result of `get()` is `None` which is invalid for the `Connection`.

Reply:
Drat, you're right. That is awkward.

How about we modify mqtt_connection_builder.py so it doesn't suffer from this problem. Like add this helper function:
```
def _get_kwarg(kwargs, name, default=None):
    """
    Returns kwargs['name'] if it exists and is not None.
    Otherwise returns default.
    """
    val = kwargs.get(name)
    if val is None:
        val = default
    return val
```

and then replace all calls in that file like `kwargs.get('name', default_value)` with `_get_kwarg(kwargs, 'name', default_value)`